src/Rolls.sol

```diff
@@ -177,33 +178,18 @@ contract Rolls is IRolls, BaseEmergencyAdmin {
         require(block.timestamp <= takerPos.expiration, "taker position expired");
 
         ShortProviderNFT providerNFT = takerPos.providerNFT;
-        uint providerId = takerPos.providerPositionId;
+        uint providerId = takerPos.providerId;
         // caller is owner
         require(msg.sender == providerNFT.ownerOf(providerId), "not provider ID owner");
 
         // sanity check bounds
-        require(minPrice < maxPrice, "max price not higher than min price");
-        require(_abs(rollFeeDeltaFactorBIPS) <= BIPS_BASE, "invalid fee delta change");
+        require(minPrice <= maxPrice, "max price lower than min price");
+        require(SignedMath.abs(rollFeeDeltaFactorBIPS) <= BIPS_BASE, "invalid fee delta change");
         require(block.timestamp <= deadline, "deadline passed");
 
         // pull the NFT
         providerNFT.transferFrom(msg.sender, address(this), providerId);
 
-        // @dev if provider expects to pay, check that they have granted sufficient balance and approvals already
-        // according to their max payment expectation
-        if (minToProvider < 0) {
-            uint maxFromProvider = uint(-minToProvider);
-            // @dev provider may still have insufficient allowance or balance when user will try to accept
-            // but this check makes it a tiny bit harder to spoof offers, and reduces chance of errors.
-            // Depositing the funds for each roll offer is avoided because it's capital inefficient, since
-            // each offer is for a specific position.
-            require(cashAsset.balanceOf(msg.sender) >= maxFromProvider, "insufficient cash balance");
```

Comment:
So it was there to reduce chance of mistakes and prevent spoofing orders. However, it seemed both not necessary and not sufficient to do either. It's not sufficient because both approval and balance can change after creating an offer, and it unnecessary because an order can already be created only by a single actor - the provider holding the NFT. Furthermore, the provider deposits their NFT into Rolls on creation, so already has a cost to doing this, because at the very least they'll need to withdraw (cancelOffer) to withdraw the provider position or do anything else with it.